### ISSUE TITLE
[Bug] [PRAxSRI] "Generating Strings enum failed, Line number for some keys could not be found" in project setup

### ISSUE THREAD

**Issue by francosolerio** [FlineDev/ReMafoX#58]
## Steps to Reproduce:

1. In the Projects Browser, press "Setup new project"
2. Drag & drop a project & wait for the analyzer to complete
3. Press "Continue" in the Quick Config screen

## Expected Behavior:

I expect the project to finish setup and show me the configuration file so I can adjust it, instead of showing me an error.

## Current Behavior:

The generation fails with this code: `PRAxSRI` and this message:

> Generating Strings enum failed with error: There is an issue with the contents of the Strings file at: ..../Castamatic/Castamatic/en.lproj/Localizable.stringsdict
> System Message: Line number for some keys could not be found.

I already tried the workarounds for issues #17 and #15 with no success: my strings files don't contain any "//" comments.

This is the content of my Localizable.stringsdict file:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Delete after &lt;color&gt;%d&lt;/color&gt; week(s)</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>Delete after %#@weeks@</string>
		<key>weeks</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>u</string>
			<key>zero</key>
			<string>&lt;color&gt;Never&lt;/color&gt; delete</string>
			<key>one</key>
			<string>&lt;color&gt;1&lt;/color&gt; week</string>
			<key>other</key>
			<string>&lt;color&gt;%u&lt;/color&gt; weeks</string>
		</dict>
	</dict>
	<key>Keep last &lt;color&gt;%d&lt;/color&gt; episode(s)</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>Keep %#@episodes_to_keep@</string>
		<key>episodes_to_keep</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>u</string>
			<key>zero</key>
			<string>&lt;color&gt;no&lt;/color&gt; episode</string>
			<key>one</key>
			<string>the &lt;color&gt;latest&lt;/color&gt; episode</string>
			<key>other</key>
			<string>last &lt;color&gt;%u&lt;/color&gt; episodes</string>
		</dict>
	</dict>
	<key>synced_podcast_count</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>%#@synced_podcast_count@</string>
		<key>synced_podcast_count</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>u</string>
			<key>zero</key>
			<string>No new podcasts synced</string>
			<key>one</key>
			<string>One new podcast synced</string>
			<key>other</key>
			<string>%u new podcasts synced</string>
		</dict>
	</dict>
	<key>%d podcast(s)</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>%#@podcast_count@</string>
		<key>podcast_count</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>u</string>
			<key>zero</key>
			<string>No podcast</string>
			<key>one</key>
			<string>1 podcast</string>
			<key>other</key>
			<string>%u podcasts</string>
		</dict>
	</dict>
	<key>Included in &lt;color&gt;%d&lt;/color&gt; playlist(s)</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>Included in %#@playlist_number@</string>
		<key>playlist_number</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>u</string>
			<key>zero</key>
			<string>&lt;color&gt;no&lt;/color&gt; playlist</string>
			<key>one</key>
			<string>&lt;color&gt;1&lt;/color&gt; playlist</string>
			<key>other</key>
			<string>&lt;color&gt;%u&lt;/color&gt; playlists</string>
		</dict>
	</dict>
	<key>%d day(s)</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>%#@day_count@</string>
		<key>day_count</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>other</key>
			<string>%u days</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>u</string>
			<key>zero</key>
			<string>Zero days</string>
			<key>one</key>
			<string>day</string>
		</dict>
	</dict>
	<key>%d week(s)</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>%#@week_count@</string>
		<key>week_count</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>u</string>
			<key>zero</key>
			<string>0 weeks</string>
			<key>one</key>
			<string>week</string>
			<key>other</key>
			<string>%u weeks</string>
		</dict>
	</dict>
	<key>%d month(s)</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>%#@month_count@</string>
		<key>month_count</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>u</string>
			<key>zero</key>
			<string>0 months</string>
			<key>one</key>
			<string>month</string>
			<key>other</key>
			<string>%u months</string>
		</dict>
	</dict>
	<key>%d year(s)</key>
	<dict>
		<key>NSStringLocalizedFormatKey</key>
		<string>%#@year_count@</string>
		<key>year_count</key>
		<dict>
			<key>NSStringFormatSpecTypeKey</key>
			<string>NSStringPluralRuleType</string>
			<key>NSStringFormatValueTypeKey</key>
			<string>u</string>
			<key>zero</key>
			<string>0 years</string>
			<key>one</key>
			<string>year</string>
			<key>other</key>
			<string>%u years</string>
		</dict>
	</dict>
</dict>
</plist>

```

## Environment

<details>
<summary>Show environment details</summary>

- App Version: 1.0.3 (Build 52)
- System Version: 12.6.1 (Build 21G217)
- System CPU: Apple M1 Max
- Tier: Free

</details>

**Comment by Jeehut** [comment]
@francosolerio Thank you for reporting this issue.

Thanks to the provided `Localizable.stringsdict` file, I could easily reproduce and fix this issue. I had a bug in the logic to escape/unescape XML String literals in the key portion such as `Delete after &lt;color&gt;%d&lt;/color&gt; week(s)`. My line number search logic was searching for `Delete after <color>%d</color> week(s)` which didn't exist in the file, leading to this error.

The fix will be part of the next release `1.1.1` which I will upload soon. I'll keep you updated!

**Comment by Jeehut** [comment]
I just submitted version `1.1.1` with the fix to the App Store. It should be available within 24 hours. 🎉
